---
id: element-newline
title: "@html-eslint/element-newline"
---

## Require newline between elements

### Rule Details

This rule enforces newline between elements.

Examples of **incorrect** code for this rule:

```html
<html>
  <head><title>newline</title></head>
</html>
```

Examples of **correct** code for this rule:

```html
<html>
  <head>
    <title>newline</title>
  </head>
</html>
```
