---
id: no-inline-styles
title: "no-inline-styles"
---

## Disallow inline styles.

### Rule Details

This rule disallow the use of inline styles.

Examples of **incorrect** code for this rule:

```html
<div style="color:#ff0a00"></div>
```

Examples of **correct** code for this rule:

```html
<div class="some-color"></div>
```

### Further reading

[MDN - id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id)
