---
id: no-restricted-attrs
title: "@html-eslint/no-restricted-attrs"
---

## Disallow specified attributes

### Rule Details


Examples of **incorrect** code for this rule:

```html
```

Examples of **correct** code for this rule:

```html
```
