---
id: no-trailing-spaces
title: "no-trailing-spaces"
---

# no-trailing-spaces

Disallow trailing whitespaces at the end of lines

## Rule Details

This rule disallows trailing whitespace (spaces, tabs) at the end of lines.

Examples of **incorrect** code for this rule:

```html
<div>••</div>
```

Examples of **correct** code for this rule:

```html
<div></div>
```
