---
id: no-target-blank
title: "@html-eslint/no-target-blank"
---

## Disallow usage of unsafe `target='_blank'`

### Rule Details

...

Examples of **incorrect** code for this rule:

```html
...
```

Examples of **correct** code for this rule:

```html
...
```

### Further reading
