---
id: require-doctype
title: "@html-eslint/require-doctype"
---

## Require `<!DOCTYPE html>` in the document

### Rule Details

This rule enforces the `<!DOCTYPE html>`.

Examples of **incorrect** code for this rule:

```html
<html></html>
```

Examples of **correct** code for this rule:

```html
<!DOCTYPE html>
<html></html>
```

## Further reading

[MDN - Doctype](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)
