# require-explicit-size

## How to use

```js,.eslintrc.js
module.exports = {
  rules: {
    "@html-eslint/require-explicit-size": "error",
  },
};
```

## Rule Details

Examples of **incorrect** code for this rule:

```html,incorrect

```

Examples of **correct** code for this rule:

```html,incorrect

```
