# Require newline between elements

## Rule Details

This rule enforce newline between elements

Examples of **incorrect** code for this rule:

<!-- prettier-ignore-start -->
```html
<html><head><title>newline</title></head></html>
```
<!-- prettier-ignore-end -->

Examples of **correct** code for this rule:

<!-- prettier-ignore-start -->
```html
<html>
<head>
<title>newline</title>
</head>
</html>
```
<!-- prettier-ignore-end -->
