# Enforce consistent indentation

Examples of **incorrect** code for this rule:

<!-- prettier-ignore-start -->
```html
<html>
<body>
</body>
</html>
```
<!-- prettier-ignore-end -->

Examples of **correct** code for this rule:

<!-- prettier-ignore-start -->
```html
<html>
    <body>
    </body>
</html>
```
<!-- prettier-ignore-end -->
