# Enforce consistent indentation

Examples of **incorrect** code for this rule:

```html
<html>
<body>
</body>
</html>
```

Examples of **correct** code for this rule:

```html
<html>
    <body>
    </body>
</html>
```
