# Enforce to use `<meta name="viewport" ...>` in the `<head></head>`.

## Rule Details

Examples of **incorrect** code for this rule:

```html
<html>
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
  </head>
</html>
```

Examples of **correct** code for this rule:

```html
<html>
  <head>
    <meta name="description" content="ESLint plugin for HTML" />
  </head>
</html>
```
