<!DOCTYPE html><html><head><meta charSet="utf-8"/><title>HTML ESLint</title><link rel="shortcut icon" href="/favicon.png"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><meta name="description" content="ESLint plugin for linting HTML. Find and fix problems in your HTML code"/><meta name="og:title" content="ESLint plugin for linting HTML."/><meta name="og:description" content="Find and fix problems in your HTML code"/><meta name="twitter:title" content="ESLint plugin for linting HTML."/><meta name="twitter:description" content="Find and fix problems in your HTML code"/><meta name="next-head-count" content="9"/><link rel="preload" href="/html-eslint/_next/static/css/23c37370c3d2aab8.css" as="style"/><link rel="stylesheet" href="/html-eslint/_next/static/css/23c37370c3d2aab8.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/html-eslint/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/html-eslint/_next/static/chunks/webpack-47a6e717ffb4af42.js" defer=""></script><script src="/html-eslint/_next/static/chunks/framework-22108eb722d84e59.js" defer=""></script><script src="/html-eslint/_next/static/chunks/main-140245b5185c1982.js" defer=""></script><script src="/html-eslint/_next/static/chunks/pages/_app-5fc430d50cf8f165.js" defer=""></script><script src="/html-eslint/_next/static/chunks/pages/docs/%5B%5B...paths%5D%5D-20c7303565bff3c6.js" defer=""></script><script src="/html-eslint/_next/static/9vfwsqTBJOSOD3en28Smg/_buildManifest.js" defer=""></script><script src="/html-eslint/_next/static/9vfwsqTBJOSOD3en28Smg/_ssgManifest.js" defer=""></script></head><body><div id="__next"><div class="min-h-[100%]"><header class="bg-slate-900 text-white flex py-3 pl-3 pr-6 items-center fixed w-full z-50 flex-col text-sm md:flex-row md:text-base"><a class="flex flex-row items-center" href="/html-eslint/"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" style="width:35px;height:35px"><path fill="#5229cb" d="M512.21 180.5h-295.6l26.64 551.73L512.21 843.5l268.96-111.27 26.64-551.73z"></path><path d="M678.89 395.46L519.3 302.19a14.032 14.032 0 00-14.17 0l-159.59 93.27a10.797 10.797 0 00-5.35 9.33v183.32c0 3.84 2.04 7.39 5.35 9.33l159.59 93.27c4.38 2.56 9.79 2.56 14.17 0l159.59-93.27c3.32-1.94 5.35-5.49 5.35-9.33V404.79c0-3.84-2.03-7.39-5.35-9.33z" fill="#fff"></path><path d="M623.16 429.23l-106.24-62.09a9.356 9.356 0 00-9.43 0l-106.24 62.09a7.178 7.178 0 00-3.56 6.21v122.03a7.2 7.2 0 003.56 6.21l106.24 62.09c2.91 1.7 6.52 1.7 9.43 0l106.24-62.09a7.178 7.178 0 003.56-6.21V435.44a7.157 7.157 0 00-3.56-6.21z" fill="#827dfa"></path></svg><h2>HTML ESLint</h2></a><nav class="w-full md:w-auto md:ml-auto"><ul class="flex gap-4 justify-around "><li><a class="hover:text-white text-slate-400" href="/html-eslint/docs/getting-started/">Getting Started</a></li><li><a class="hover:text-white text-white" href="/html-eslint/docs/rules/">Rules</a></li><li><a class="hover:text-white text-slate-400" href="/html-eslint/playground/">Playground</a></li><li><a class="hover:text-white text-slate-400" href="https://github.com/yeonjuan/html-eslint">GitHub</a></li></ul></nav></header><main class="pt-[60px] px-2 md:px-8" style="min-height:calc(100vh - 132px)"><div class="relative md:flex"><aside class="hidden sticky overflow-y-auto top-[60px] py-4 flex-[0_0_240px] border-r-slate-300 border-r-[1px] md:block" style="height:calc(100vh - 60px)"><ul class="pl-1 pb-1"><li><h1 class="my-1 text-lg text-slate-900">Getting Started</h1><ul class="pl-2 pb-1"><li><h2 class="my-1 text-base text-slate-800">Disabling Rules</h2></li><li><h2 class="my-1 text-base text-slate-800">CLI</h2></li></ul></li><li><h1 class="my-1 text-lg text-slate-900">Rules</h1><ul class="pl-2 pb-1"><li><h2 class="my-1 text-base text-slate-800">Best Practice</h2><ul class="pl-3 pb-1"><li><h3 class="my-1 text-sm text-slate-600">no-duplicate-attrs</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-duplicate-id</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-inline-styles</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-obsolete-tags</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-restricted-attrs</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-target-blank</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-button-type</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-closing-tags</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-doctype</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-li-container</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-meta-charset</h3></li></ul></li><li><h2 class="my-1 text-base text-slate-800">SEO</h2><ul class="pl-3 pb-1"><li><h3 class="my-1 text-sm text-slate-600">no-multiple-h1</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-lang</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-meta-description</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-title</h3></li></ul></li><li><h2 class="my-1 text-base text-slate-800">Accessibility</h2><ul class="pl-3 pb-1"><li><h3 class="my-1 text-sm text-slate-600">no-abstract-roles</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-accesskey-attrs</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-aria-hidden-body</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-non-scalable-viewport</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-positive-tabindex</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-skip-heading-levels</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-frame-title</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-img-alt</h3></li><li><h3 class="my-1 text-sm text-slate-600">require-meta-viewport</h3></li></ul></li><li><h2 class="my-1 text-base text-slate-800">Style</h2><ul class="pl-3 pb-1"><li><h3 class="my-1 text-sm text-slate-600">element-newline</h3></li><li><h3 class="my-1 text-sm text-slate-600">id-naming-convention</h3></li><li><h3 class="my-1 text-sm text-slate-600">indent</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-extra-spacing-attrs</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-multiple-empty-lines</h3></li><li><h3 class="my-1 text-sm text-slate-600">no-trailing-spaces</h3></li><li><h3 class="my-1 text-sm text-slate-600">quotes</h3></li></ul></li></ul></li></ul></aside><article class="markdown-body"><h1>require-meta-charset</h1>
<p>Enforce to use <code>&#x3C;meta charset="..."></code> in the <code>&#x3C;head>&#x3C;/head></code>.</p>
<h2>How to use</h2>
<p>.eslintrc.js</p>
<pre><code class="hljs language-js"><span class="hljs-variable language_">module</span>.<span class="hljs-property">exports</span> = {
  <span class="hljs-attr">rules</span>: {
    <span class="hljs-string">"@html-eslint/require-meta-charset"</span>: <span class="hljs-string">"error"</span>,
  },
};
</code></pre>
<h2>Rule Details</h2>
<p>Examples of <strong>incorrect</strong> code for this rule:</p>
<pre><code class="hljs language-html"><span class="hljs-tag">&#x3C;<span class="hljs-name">html</span>></span>
  <span class="hljs-tag">&#x3C;<span class="hljs-name">head</span>></span> <span class="hljs-tag">&#x3C;/<span class="hljs-name">head</span>></span>
<span class="hljs-tag">&#x3C;/<span class="hljs-name">html</span>></span>
</code></pre>
<p>Examples of <strong>correct</strong> code for this rule:</p>
<pre><code class="hljs language-html"><span class="hljs-tag">&#x3C;<span class="hljs-name">html</span>></span>
  <span class="hljs-tag">&#x3C;<span class="hljs-name">head</span>></span>
    <span class="hljs-tag">&#x3C;<span class="hljs-name">meta</span> <span class="hljs-attr">charset</span>=<span class="hljs-string">"UTF-8"</span> /></span>
  <span class="hljs-tag">&#x3C;/<span class="hljs-name">head</span>></span>
<span class="hljs-tag">&#x3C;/<span class="hljs-name">html</span>></span>
</code></pre></article></div></main><footer class="bg-slate-900 text-slate-400 py-4 px-4 md:px-16"><section class="flex justify-between my-4"><div class="flex-1"><h5 class="text-white text-sm">Docs</h5><div class="pl-1"><ul><li><a class="hover:text-white text-slate-400 text-xs" href="/html-eslint/docs/getting-started/">Getting Started</a></li><li><a class="hover:text-white text-slate-400 text-xs" href="/html-eslint/docs/rules/">Rules</a></li></ul></div></div><div class="flex-1"><h5 class="text-white text-sm">Playground</h5><div class="pl-1"><a class="hover:text-white text-slate-400 text-xs" href="/html-eslint/playground/">Playground</a></div></div><div class="flex-1"><h5 class="text-white text-sm">More</h5><div class="pl-1"><a class="hover:text-white text-slate-400 text-xs" href="/html-eslint/docs/getting-started/">GitHub</a></div></div></section></footer></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"html":"\u003ch1\u003erequire-meta-charset\u003c/h1\u003e\n\u003cp\u003eEnforce to use \u003ccode\u003e\u0026#x3C;meta charset=\"...\"\u003e\u003c/code\u003e in the \u003ccode\u003e\u0026#x3C;head\u003e\u0026#x3C;/head\u003e\u003c/code\u003e.\u003c/p\u003e\n\u003ch2\u003eHow to use\u003c/h2\u003e\n\u003cp\u003e.eslintrc.js\u003c/p\u003e\n\u003cpre\u003e\u003ccode class=\"hljs language-js\"\u003e\u003cspan class=\"hljs-variable language_\"\u003emodule\u003c/span\u003e.\u003cspan class=\"hljs-property\"\u003eexports\u003c/span\u003e = {\n  \u003cspan class=\"hljs-attr\"\u003erules\u003c/span\u003e: {\n    \u003cspan class=\"hljs-string\"\u003e\"@html-eslint/require-meta-charset\"\u003c/span\u003e: \u003cspan class=\"hljs-string\"\u003e\"error\"\u003c/span\u003e,\n  },\n};\n\u003c/code\u003e\u003c/pre\u003e\n\u003ch2\u003eRule Details\u003c/h2\u003e\n\u003cp\u003eExamples of \u003cstrong\u003eincorrect\u003c/strong\u003e code for this rule:\u003c/p\u003e\n\u003cpre\u003e\u003ccode class=\"hljs language-html\"\u003e\u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;\u003cspan class=\"hljs-name\"\u003ehtml\u003c/span\u003e\u003e\u003c/span\u003e\n  \u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;\u003cspan class=\"hljs-name\"\u003ehead\u003c/span\u003e\u003e\u003c/span\u003e \u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;/\u003cspan class=\"hljs-name\"\u003ehead\u003c/span\u003e\u003e\u003c/span\u003e\n\u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;/\u003cspan class=\"hljs-name\"\u003ehtml\u003c/span\u003e\u003e\u003c/span\u003e\n\u003c/code\u003e\u003c/pre\u003e\n\u003cp\u003eExamples of \u003cstrong\u003ecorrect\u003c/strong\u003e code for this rule:\u003c/p\u003e\n\u003cpre\u003e\u003ccode class=\"hljs language-html\"\u003e\u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;\u003cspan class=\"hljs-name\"\u003ehtml\u003c/span\u003e\u003e\u003c/span\u003e\n  \u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;\u003cspan class=\"hljs-name\"\u003ehead\u003c/span\u003e\u003e\u003c/span\u003e\n    \u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;\u003cspan class=\"hljs-name\"\u003emeta\u003c/span\u003e \u003cspan class=\"hljs-attr\"\u003echarset\u003c/span\u003e=\u003cspan class=\"hljs-string\"\u003e\"UTF-8\"\u003c/span\u003e /\u003e\u003c/span\u003e\n  \u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;/\u003cspan class=\"hljs-name\"\u003ehead\u003c/span\u003e\u003e\u003c/span\u003e\n\u003cspan class=\"hljs-tag\"\u003e\u0026#x3C;/\u003cspan class=\"hljs-name\"\u003ehtml\u003c/span\u003e\u003e\u003c/span\u003e\n\u003c/code\u003e\u003c/pre\u003e","trees":[{"doc":{"title":"Getting Started","markdown":"\n# Getting Started\n\n## Prerequisite\n\n- Node.js `\u003e=8.10.0`.\n- ESLint `\u003e=6`.\n\n## Installation\n\n```\n$ npm install --save-dev eslint @html-eslint/parser @html-eslint/eslint-plugin\n```\n\n## Configuration\n\nPopulate it with the following on your `.eslintrc.js`. If it does not exist create a `.eslintrc.js` config file in the root of your project.\nWe can apply these [plugin rules](#Rules) to only HTML files(`*.html`) by using `overrides` in `.eslintrc.js`. (see [ESLint Configuration](https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns))\n\n```js\nmodule.exports = {\n  //...\n  plugins: [\"@html-eslint\"],\n  overrides: [\n    {\n      files: [\"*.html\"],\n      parser: \"@html-eslint/parser\",\n      extends: [\"plugin:@html-eslint/recommended\"],\n    },\n  ],\n};\n```\n\n## Editor Configuration\n\n### VSCode\n\nTo get [vscode-eslint](https://github.com/microsoft/vscode-eslint) support, we need to add the following in vscode settings.\n\n- `.vscode/settings.json`:\n\n```json\n{\n  \"eslint.enable\": true,\n  \"eslint.validate\": [\n    \"javascript\", // ...\n    \"html\" // Add \"html\" to enable linting `.html` files.\n  ]\n}\n```\n\n## Recommended Configs\n\nThis plugin provides `plugin:@html-eslint/recommended`. The recommended configuration contains the rules marked ‚≠ê in [Rules](#Rules).\n","path":"/docs/getting-started"},"children":[{"doc":{"title":"Disabling Rules","markdown":"\n# Disabling Rules\n\n## Disable rules with inline comments.\n\nTo disable rule in a file, use HTML comments (`\u003c!-- ... --\u003e`) in the following format:\n\nDisable all rules in an entire file:\n\n```html\n\u003c!-- eslint-disable --\u003e\n\u003cdiv foo=\"foo\" foo=\"foo\"\u003e\u003c/div\u003e\n```\n\nDisable or enable specific rules in an entire file:\n\n```html\n\u003c!-- eslint-disable @html-eslint/no-duplicate-attrs --\u003e\n\u003cdiv foo=\"foo\" foo=\"foo\"\u003e\u003c/div\u003e\n\n\u003c!-- eslint-enable @html-eslint/no-duplicate-attrs  --\u003e\n\u003cdiv foo=\"foo\" foo=\"foo\"\u003e\u003c/div\u003e\n```\n\nDisable all rules on a specific line:\n\n```html\n\u003c!-- eslint-disable-next-line --\u003e\n\u003cdiv foo=\"foo\" foo=\"foo\"\u003e\u003c/div\u003e\n```\n\nDisable a specific rule on a specific line:\n\n```html\n\u003c!-- eslint-disable-next-line @html-eslint/no-duplicate-attrs --\u003e\n\u003cdiv foo=\"foo\" foo=\"foo\"\u003e\u003c/div\u003e\n```\n","path":"/docs/disabling-rules"}},{"doc":{"title":"CLI","markdown":"\n# CLI\n\n## Installation\n\n```\n$ npm install -g @html-eslint/cli\n```\n\n## Configuration\n\n- `.htmleslintrc.js`\n\n```javascript\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-lang\": \"error\",\n    \"@html-eslint/require-img-alt\": \"error\",\n    \"@html-eslint/require-doctype\": \"error\",\n    \"@html-eslint/require-title\": \"error\",\n    \"@html-eslint/no-multiple-h1\": \"error\",\n    // ...\n  },\n};\n```\n\n## Run\n\n```\n$ html-eslint https://www.example.com\n```\n\n## Options\n\n- `--config` (`-c`)\n\n  Specify configuration file to use.\n\n  ```\n  $ html-eslint https://www.example.com --config path/of/configuration\n  # or\n  $ html-eslint https://www.example.com -c path/of/configuration\n  ```\n\n- `--check-style`\n\n  Enable `Style` rules in configuration. By default, All rules with the category `Style` are ignored.\n\n  ```\n  $ html-eslint https://www.example.com --check-style\n  ```\n","path":"/docs/cli"}}]},{"doc":{"title":"Rules","markdown":"\n# Rules\n\n- üîß - Meaning the rule can fix problems automatically by running eslint `--fix` options.\n- ‚≠ê - Meaning the rule is recommended.\n\n### Best Practice\n\n| rule                                                            | description                                                |      |\n| :-------------------------------------------------------------- | :--------------------------------------------------------- | :--- |\n| [@html-eslint/require-doctype](rules/require-doctype)           | Require `\u003c!DOCTYPE HTML\u003e` in html,                         | ‚≠êüîß |\n| [@html-eslint/no-duplicate-id](rules/no-duplicate-id)           | Disallow to use duplicate id                               | ‚≠ê   |\n| [@html-eslint/no-inline-styles](rules/no-inline-styles)         | Disallow using inline style                                |      |\n| [@html-eslint/require-li-container](rules/require-li-container) | Enforce `\u003cli\u003e` to be in `\u003cul\u003e`, `\u003col\u003e` or `\u003cmenu\u003e`.        | ‚≠ê   |\n| [@html-eslint/no-obsolete-tags](rules/no-obsolete-tags)         | Disallow to use obsolete elements in HTML5                 | ‚≠ê   |\n| [@html-eslint/require-closing-tags](rules/require-closing-tags) | Require closing tags.                                      | ‚≠êüîß |\n| [@html-eslint/require-meta-charset](rules/require-meta-charset) | Enforce to use `\u003cmeta charset=\"...\"\u003e` in `\u003chead\u003e`          |      |\n| [@html-eslint/no-target-blank](rules/no-target-blank)           | Disallow usage of unsafe `target='_blank'`                 |      |\n| [@html-eslint/no-duplicate-attrs](rules/no-duplicate-attrs)     | Disallow to use duplicate attributes                       | ‚≠ê   |\n| [@html-eslint/require-button-type](rules/require-button-type)   | Require use of button element with a valid type attribute. |      |\n| [@html-eslint/no-restricted-attrs](rules/no-restricted-attrs)   | Disallow specified attributes                              |      |\n\n### SEO\n\n| rule                                                                    | description                                            |     |\n| :---------------------------------------------------------------------- | :----------------------------------------------------- | :-- |\n| [@html-eslint/require-lang](rules/require-lang)                         | Require `lang` attribute at `\u003chtml\u003e` tag               | ‚≠ê  |\n| [@html-eslint/require-title](rules/require-title)                       | Require `\u003ctitle\u003e\u003ctitle/\u003e` in the `\u003chead\u003e\u003chead/\u003e`       | ‚≠ê  |\n| [@html-eslint/no-multiple-h1](rules/no-multiple-h1)                     | Disallow multiple `\u003ch1\u003e\u003c/h1\u003e`.                         | ‚≠ê  |\n| [@html-eslint/require-meta-description](rules/require-meta-description) | Require use of `\u003cmeta name=\"description\"\u003e` in `\u003chead\u003e` |     |\n\n### Accessibility\n\n| rule                                                                    | description                                                     |     |\n| :---------------------------------------------------------------------- | :-------------------------------------------------------------- | :-- |\n| [@html-eslint/require-img-alt](rules/require-img-alt)                   | Require `alt` attribute at `\u003cimg\u003e` tag                          | ‚≠ê  |\n| [@html-eslint/no-skip-heading-levels](rules/no-skip-heading-levels)     | Disallow skipping heading levels                                |     |\n| [@html-eslint/require-frame-title](rules/require-frame-title)           | Require `title` in `\u003cframe\u003e`, `\u003ciframe\u003e`                        |     |\n| [@html-eslint/no-non-scalable-viewport](rules/no-non-scalable-viewport) | Disallow use of `user-scalable=no` in `\u003cmeta name=\"viewport\"\u003e`. |     |\n| [@html-eslint/no-positive-tabindex](rules/no-positive-tabindex)         | Disallow use of positive `tabindex`.                            |     |\n| [@html-eslint/require-meta-viewport](rules/require-meta-viewport)       | Enforce to use `\u003cmeta name=\"viewport\"\u003e` in `\u003chead\u003e`             |     |\n| [@html-eslint/no-abstract-roles](rules/no-abstract-roles)               | Disallow to use of abstract roles                               |     |\n| [@html-eslint/no-aria-hidden-body](rules/no-aria-hidden-body)           | Disallow to use aria-hidden attributes on the `body` element.   |     |\n| [@html-eslint/no-accesskey-attrs](rules/no-accesskey-attrs)             | Disallow to use of accesskey attribute                          |     |\n\n### Styles\n\n| rule                                                                  | description                                                       |      |\n| :-------------------------------------------------------------------- | :---------------------------------------------------------------- | :--- |\n| [@html-eslint/no-extra-spacing-attrs](rules/no-extra-spacing-attrs)   | Disallow an extra spacing around attributes                       | ‚≠êüîß |\n| [@html-eslint/element-newline](rules/element-newline)                 | Enforce newline between elements.                                 | ‚≠êüîß |\n| [@html-eslint/indent](rules/indent)                                   | Enforce consistent indentation                                    | ‚≠êüîß |\n| [@html-eslint/quotes](rules/quotes)                                   | Enforce consistent quoting attributes with double(\") or single(') | ‚≠êüîß |\n| [@html-eslint/id-naming-convention](rules/id-naming-convention)       | Enforce consistent naming id attributes                           |      |\n| [@html-eslint/no-multiple-empty-lines](rules/no-multiple-empty-lines) | Disallow multiple empty lines                                     | üîß   |\n| [@html-eslint/no-trailing-spaces](rules/no-trailing-spaces)           | Disallow trailing whitespace at the end of lines                  | üîß   |\n","path":"/docs/rules"},"children":[{"title":"Best Practice","children":[{"doc":{"title":"no-duplicate-attrs","markdown":"\n# no-duplicate-attrs\n\nDisallow duplicate attributes.\n\n## How to use\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-duplicate-attrs\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of duplicate attributes.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv foo=\"foo1\" foo=\"foo2\"\u003e\u003c/div\u003e\n\u003cdiv foo foo\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv foo=\"foo\"\u003e\u003c/div\u003e\n\u003cdiv bar\u003e\u003c/div\u003e\n```\n","path":"/docs/rules/no-duplicate-attrs"}},{"doc":{"title":"no-duplicate-id","markdown":"\n# no-duplicate-id\n\nDisallow duplicate `id` attributes.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-duplicate-id\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallow the use of duplicate `id` attributes.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n\u003cdiv id=\"bar\"\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n- [MDN - id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id)\n","path":"/docs/rules/no-duplicate-id"}},{"doc":{"title":"no-inline-styles","markdown":"\n# no-inline-styles\n\nDisallow inline styles.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-inline-styles\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of inline styles.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv style=\"color:#ff0a00\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv class=\"some-color\"\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n- [MDN - id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id)\n","path":"/docs/rules/no-inline-styles"}},{"doc":{"title":"no-obsolete-tags","markdown":"\n# no-obsolete-tags\n\nDisallow using obsolete tags in HTML5\n\nThe following element list is obsoleted in HTML5.\nIt's not encouraged to use these tags.\n\n```\napplet, acronym, bgsound, dir, frame, frameset, noframes, isindex, keygen, listing, menuitem, nextid, noembed, plaintext, rb, rtc, strike, xmp, basefont, big, blink, center, font, marquee, multicol, nobr, spacer, tt\n```\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-obsolete-tags\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallow using obsolete tags.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003capplet\u003e\u003c/applet\u003e\n\u003cdir\u003e\u003c/dir\u003e\n```\n\n## Further reading\n\n- [html spec 16.2. Non-conforming features](https://html.spec.whatwg.org/#non-conforming-features)\n","path":"/docs/rules/no-obsolete-tags"}},{"doc":{"title":"no-restricted-attrs","markdown":"\n# no-restricted-attrs\n\nDisallow specified attributes\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    '@html-eslint/no-restricted-attrs': [\"error\",  {\n  tagPatterns: [\"^div$\", \"^img$\"],\n    attrPatterns: [\"data-.*\"]\n    message: \"\\'data-x\\' is restricted.\"\n  }]\n  }\n};\n```\n\n## Rule Details\n\nThis rule allows you to specify attributes that you don't want to use in your application.\n\n### Options\n\nThis rule takes an array of option objects, where the `tagPatterns` and `attrPatterns` are specified.\n\n- `tagPatterns`: an array of strings representing regular expression pattern, disallows tag name that match any of the patterns.\n- `attrPatterns`: an array of strings representing regular expression pattern, disallows attribute name that match any of the patterns.\n- `message` (optional): a string for custom message.\n\n```js\nmodule.exports = {\n  \"rules\": {\n    \"@html-eslint/no-restricted-attrs\": [\n      \"error\",\n      {\n        tagPatterns: [\"^div$\", \"^img$\"],\n        attrPatterns: [\"data-.*\"]\n        message: \"\\'data-x\\' is restricted.\"\n      },\n      {\n        tagPatterns: [\"^img$\"],\n        attrPatterns: [\"^alt$\"]\n        message: \"\\'alt\\' is restricted.\"\n      }\n    ],\n  }\n};\n```\n\nExamples of **incorrect** code for this rule with the option below:\n\n```json\n{\n  \"tagPatterns\": [\"^img$\", \"^div$\"],\n  \"attrPatterns\": [\"^data-.*\"],\n  \"message\": \"Do not use data-* attr\"\n}\n```\n\n```html\n\u003cdiv data-name=\"foo\"\u003e\u003c/div\u003e\n\u003cimg data-name=\"foo\"\u003e\u003c/div\u003e\n```\n","path":"/docs/rules/no-restricted-attrs"}},{"doc":{"title":"no-target-blank","markdown":"\n# no-target-blank\n\nDisallow usage of unsafe `target='_blank'`\n\n## How to use\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-target-blank\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003ca target=\"_blank\" href=\"http://example.com/\"\u003e\u003c/a\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003ca target=\"_blank\" href=\"relative/path/\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noreferrer\" href=\"http://example.com/\"\u003e\u003c/a\u003e\n```\n","path":"/docs/rules/no-target-blank"}},{"doc":{"title":"require-button-type","markdown":"\n# require-button-type\n\nRequire use of button element with a valid type attribute.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-button-type\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces use of a valid type attribute for button elements. (`\"button\"`, `\"submit\"`, `\"reset\"`)\n\nExamples of **incorrect** code for this rule:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cbutton\u003e\u003c/button\u003e\n\u003cbutton type=\"invalid\"\u003e\u003c/button\u003e\n```\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cbutton type=\"submit\"\u003e\u003c/button\u003e\n\u003cbutton type=\"button\"\u003e\u003c/button\u003e\n\u003cbutton type=\"reset\"\u003e\u003c/button\u003e\n```\n\n## Further reading\n\n- [HTML spec - the button element](https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type)\n","path":"/docs/rules/require-button-type"}},{"doc":{"title":"require-closing-tags","markdown":"\n# require-closing-tags\n\nRequire use of closing tag.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-closing-tags\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule checks whether the tag has closing tag or not.\n\nExamples of **incorrect** code for this rule:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cdiv\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for this rule:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cdiv\u003e\u003c/div\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n### Options\n\nThis rule has an object option for [Void Elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements).\n\n- `\"selfClosing\": \"never\"`: (default) disallow using self closing tag on [Void Elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements).\n\n- `\"selfClosing\": \"always\"`: enforce using self closing tag on [Void Elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements).\n\n- `\"allowSelfClosingCustom\": false`: (default) disallow self-closing for the custom tags.\n\n- `\"allowSelfClosingCustom\": true`: allow self-closing for the custom tags.\n\n#### selfClosing : \"never\"\n\nExamples of **incorrect** code for the `{ \"selfClosing\": \"never\"}` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg /\u003e\n\u003cbase /\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for the `{ \"selfClosing\": \"never\"}` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg\u003e\n\u003cbase\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n#### selfClosing : \"always\"\n\nExamples of **incorrect** code for the `{ \"selfClosing\": \"always\" }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg\u003e\n\u003cbase\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for the `{ \"selfClosing\": \"always\" }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg /\u003e\n\u003cbase /\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n#### \"allowSelfClosingCustom\": false\n\nExamples of **incorrect** code for the `{ \"allowSelfClosingCustom\": false }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003ccustom-tag /\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for the `{ \"allowSelfClosingCustom\": false }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003ccustom-tag\u003e \u003c/custom-tag\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n#### \"allowSelfClosingCustom\": true\n\nExamples of **correct** code for the `{ \"allowSelfClosingCustom\": true }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003c!-- both allowed --\u003e\n\n\u003ccustom-tag\u003e \u003c/custom-tag\u003e\n\u003ccustom-tag /\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n## Further reading\n\n- [Void Elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements)\n","path":"/docs/rules/require-closing-tags"}},{"doc":{"title":"require-doctype","markdown":"\n# require-doctype\n\nRequire `\u003c!DOCTYPE html\u003e` in the document\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-doctype\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces the `\u003c!DOCTYPE html\u003e`.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003c!DOCTYPE html\u003e\n\u003chtml\u003e\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - Doctype](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)\n","path":"/docs/rules/require-doctype"}},{"doc":{"title":"require-li-container","markdown":"\n# require-li-container\n\nEnforce `\u003cli\u003e` to be in `\u003cul\u003e`, `\u003col\u003e` or `\u003cmenu\u003e`.\n\nThe `\u003cli\u003e` tag should be contained in a parent element: `\u003col\u003e`, `\u003cul\u003e` or `\u003cmenu\u003e`.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-li-container\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv\u003e\n  \u003cli\u003eitem 1\u003c/li\u003e\n  \u003cli\u003eitem 2\u003c/li\u003e\n\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003col\u003e\n  \u003cli\u003eitem 1\u003c/li\u003e\n  \u003cli\u003eitem 2\u003c/li\u003e\n\u003c/ol\u003e\n\n\u003cul\u003e\n  \u003cli\u003eitem 1\u003c/li\u003e\n  \u003cli\u003eitem 2\u003c/li\u003e\n\u003c/ul\u003e\n```\n\n## Further reading\n\n- [MDN - li](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li)\n","path":"/docs/rules/require-li-container"}},{"doc":{"title":"require-meta-charset","markdown":"\n# require-meta-charset\n\nEnforce to use `\u003cmeta charset=\"...\"\u003e` in the `\u003chead\u003e\u003c/head\u003e`.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-meta-charset\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e \u003c/head\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta charset=\"UTF-8\" /\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n","path":"/docs/rules/require-meta-charset"}}]},{"title":"SEO","children":[{"doc":{"title":"no-multiple-h1","markdown":"\n# no-multiple-h1\n\nDisallow multiple `\u003ch1\u003e\u003ch1\u003e`.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-multiple-h1\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of multiple `\u003ch1\u003e\u003c/h1\u003e` tags.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\n    \u003ch1\u003ehead\u003c/h1\u003e\n    \u003ch1\u003ehead\u003c/h1\u003e\n  \u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\n    \u003ch1\u003ehead\u003c/h1\u003e\n  \u003c/body\u003e\n\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - heading elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements)\n","path":"/docs/rules/no-multiple-h1"}},{"doc":{"title":"require-lang","markdown":"\n# require-lang\n\nRequire `lang` attribute at `html` tag\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-lang\": \"error\",\n  },\n};\n```\n\nThe `lang` attribute is used to represent the language used in the content.\nWe can give information to the search engine about the language used in the content.\n\n## Rule Details\n\nThis rule enforces the `lang` attribute at `html` tag\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  ...\n\u003c/html\u003e\n\n\u003chtml lang=\"\"\u003e\n  ...\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml lang=\"ko\"\u003e\n  ...\n\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - lang](https://developer.mozilla.org/ko/docs/Web/HTML/Global_attributes/lang)\n","path":"/docs/rules/require-lang"}},{"doc":{"title":"require-meta-description","markdown":"\n# require-meta-description\n\nEnforce to use `\u003cmeta name=\"description\" ...\u003e` in the `\u003chead\u003e\u003c/head\u003e`.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-meta-description\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"author\" content=\"YeonJuAn\" /\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"description\" content=\"ESLint plugin for HTML\" /\u003e\n    \u003cmeta name=\"author\" content=\"YeonJuAn\" /\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n","path":"/docs/rules/require-meta-description"}},{"doc":{"title":"require-title","markdown":"\n# require-title\n\nRequire `\u003ctitle\u003e` in the `\u003chead\u003e`.\n\n`\u003ctitle\u003e\u003ctitle/\u003e` tag is used to define the document's title.\nThe content of the title is used by the search engine to decide the order of search results.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-title\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces `\u003ctitle\u003e\u003ctitle/\u003e` tag in the `\u003chead\u003e\u003chead/\u003e`.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e \u003c/head\u003e\n\u003c/html\u003e\n\n\u003c!-- empty title --\u003e\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003ctitle\u003e \u003c/title\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003ctitle\u003eSite Title\u003c/title\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - title](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/title)\n","path":"/docs/rules/require-title"}}]},{"title":"Accessibility","children":[{"doc":{"title":"no-abstract-roles","markdown":"\n# no-abstract-roles\n\nDisallow use of abstract roles.\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-abstract-roles\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of abstract roles.\n\n- Abstract roles\n  - command\n  - composite\n  - input\n  - landmark\n  - range\n  - roletype\n  - section\n  - sectionhead\n  - select\n  - structure\n  - widget\n  - window\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv role=\"command\"\u003e\u003c/div\u003e\n\u003cdiv role=\"composite\"\u003e\u003c/div\u003e\n\u003cdiv role=\"input\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv role=\"button\"\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n- [HTML spec - Abstract Roles](https://www.w3.org/TR/wai-aria-1.0/roles#abstract_roles)\n","path":"/docs/rules/no-abstract-roles"}},{"doc":{"title":"no-accesskey-attrs","markdown":"\n# no-accesskey-attrs\n\nDisallow accesskey attributes.\n\nAccess keys are HTML attributes that allow web developers to assign keyboard shortcuts to elements. Inconsistencies between keyboard shortcuts and keyboard commands used by screenreader and keyboard only users create accessibility complications so to avoid complications, access keys should not be used.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-accesskey-attrs\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of `accesskey` attributes.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv accesskey=\"h\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n- [MDN - accesskey](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/accesskey)\n","path":"/docs/rules/no-accesskey-attrs"}},{"doc":{"title":"no-aria-hidden-body","markdown":"\n# no-aria-hidden-body\n\nDisallow to use aria-hidden attributes on the `body` element.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-aria-hidden-body\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of aria-hidden attributes on the `body` element.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cbody aria-hidden\u003e\n  \u003cbody aria-hidden=\"true\"\u003e\u003c/body\u003e\n\u003c/body\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cbody\u003e\u003c/body\u003e\n```\n\n## Further reading\n\n- [MDN - Using the aria-hidden attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute)\n","path":"/docs/rules/no-aria-hidden-body"}},{"doc":{"title":"no-non-scalable-viewport","markdown":"\n# no-non-scalable-viewport\n\nDisallow use of `user-scalable=no` in `\u003cmeta name=\"viewport\"\u003e`.\n\nThe `user-scalable=no` disables zooming on a page. It makes users with partial vision or low vision hard to read web content.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-non-scalable-viewport\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallow use of `user-scalable-no` in `\u003cmeta name=\"viewport\"\u003e`.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"viewport\" content=\"width=device-width, user-scalable=no\" /\u003e\n  \u003c/head\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"viewport\" content=\"width=device-width\" /\u003e\n  \u003c/head\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - Viewport meta tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag)\n","path":"/docs/rules/no-non-scalable-viewport"}},{"doc":{"title":"no-positive-tabindex","markdown":"\n# no-positive-tabindex\n\nDisallow use of positive `tabindex` attribute.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-positive-tabindex\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows use of positive `tabindex` attribute.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cspan tabindex=\"1\"\u003efoo\u003c/span\u003e \u003cspan tabindex=\"3\"\u003ebar\u003c/span\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cspan tabindex=\"0\"\u003efoo\u003c/span\u003e \u003cspan tabindex=\"-1\"\u003ebar\u003c/span\u003e\n```\n","path":"/docs/rules/no-positive-tabindex"}},{"doc":{"title":"no-skip-heading-levels","markdown":"\n# no-skip-heading-levels\n\nDisallow skipping heading levels.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-skip-heading-levels\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows skipping heading levels.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\n    \u003ch1\u003ehead1\u003c/h1\u003e\n    \u003ch3\u003ehead3\u003c/h3\u003e\n  \u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\n    \u003ch1\u003ehead1\u003c/h1\u003e\n    \u003ch2\u003ehead2\u003c/h2\u003e\n  \u003c/body\u003e\n\u003c/html\u003e\n```\n\n## Further reading\n\n- [MDN - heading elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements)\n","path":"/docs/rules/no-skip-heading-levels"}},{"doc":{"title":"require-frame-title","markdown":"\n# require-frame-title\n\nRequire `title` attribute in `\u003cframe\u003e` and `\u003ciframe\u003e`\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-frame-title\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces usr of `title` attribute in `\u003cframe\u003e` and `\u003ciframe\u003e`.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003ciframe src=\"...\"\u003e\u003c/iframe\u003e \u003ciframe src=\"...\" title=\"\"\u003e\u003c/iframe\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003ciframe src=\"...\" title=\"frame title\"\u003e\u003c/iframe\u003e\n```\n","path":"/docs/rules/require-frame-title"}},{"doc":{"title":"require-img-alt","markdown":"\n# require-img-alt\n\nRequire `alt` attribute at `img` tag\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-img-alt\": \"error\",\n  },\n};\n```\n\nThe screen reader uses `alt` attributes in `img` tag for describing the image.\nAnd the `alt` value is displayed if fails to load the image.\nSo the `img` tag should contain the `alt` attribute for those who cannot see images.\n\n## Rule Details\n\nThis rule enforces the `alt` attribute at `img` tag.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cimg src=\"image.png\" /\u003e\n\n\u003cimg src=\"image.png\" alt=\"\" /\u003e\n\u003c!-- empty value --\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cimg src=\"image.png\" alt=\"some description\" /\u003e\n```\n\n## Further reading\n\n- [MDN - img](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img)\n","path":"/docs/rules/require-img-alt"}},{"doc":{"title":"require-meta-viewport","markdown":"\n# require-meta-viewport\n\nEnforce to use `\u003cmeta name=\"viewport\" ...\u003e` in the `\u003chead\u003e\u003c/head\u003e`.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/require-meta-viewport\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"viewport\" content=\"width=device-width, initial-scale=1\" /\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"description\" content=\"ESLint plugin for HTML\" /\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n","path":"/docs/rules/require-meta-viewport"}}]},{"title":"Style","children":[{"doc":{"title":"element-newline","markdown":"\n# element-newline\n\nEnforce newline between elements\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/element-newline\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces newline between elements.\n\nExamples of **incorrect** code for this rule:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\u003ctitle\u003enewline\u003c/title\u003e\u003c/head\u003e\n\u003c/html\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003ctitle\u003enewline\u003c/title\u003e\n  \u003c/head\u003e\n\u003c/html\u003e\n```\n","path":"/docs/rules/element-newline"}},{"doc":{"title":"id-naming-convention","markdown":"\n# id-naming-convention\n\nEnforce naming conventions for id attributes.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/id-naming-convention\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces naming conventions for id attributes.\nIt supports 4 naming cases. `camelCase`, `snake_case`, `PascalCase`, `kebab-case` (default `snake_case`).\n\n### Options\n\n- `\"snake_case\"` (default): Enforce snake_case format.\n- `\"camelCase\"`: Enforce camelCase format.\n- `\"PascalCase\"`: Enforce PascalCase format.\n- `\"kebab-case\"`: Enforce kebab-case format.\n\n#### \"snake_case\" (default)\n\nExamples of **incorrect** code for this rule with the default `\"snake_case\"` option:\n\n```html\n\u003cdiv id=\"Foo\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the default `\"snake_case\"` option:\n\n```html\n\u003cdiv id=\"foo_bar\"\u003e\u003c/div\u003e\n```\n\n#### \"camelCase\"\n\nExamples of **incorrect** code for this rule with the `\"camelCase\"` option:\n\n```html\n\u003cdiv id=\"foo_bar\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the `\"camelCase\"` option:\n\n```html\n\u003cdiv id=\"fooBar\"\u003e\u003c/div\u003e\n```\n\n#### \"PascalCase\"\n\nExamples of **incorrect** code for this rule with the `\"PascalCase\"` option:\n\n```html\n\u003cdiv id=\"foo_bar\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the `\"PascalCase\"` option:\n\n```html\n\u003cdiv id=\"FooBar\"\u003e\u003c/div\u003e\n```\n\n### \"kebab-case\"\n\nExamples of **incorrect** code for this rule with the `\"kebab-case\"` option:\n\n```html\n\u003cdiv id=\"foo_bar\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the `\"kebab-case\"` option:\n\n```html\n\u003cdiv id=\"foo-bar\"\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n[Wiki - Naming convention](\u003chttps://en.wikipedia.org/wiki/Naming_convention_(programming)\u003e)\n","path":"/docs/rules/id-naming-convention"}},{"doc":{"title":"indent","markdown":"\n# indent\n\nEnforce consistent indentation\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/indent\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces consistent indentation styles. The default indent is `4 spaces`.\n\n### Options\n\nThis rule has two options\n\n- `number(0, 1, ..)` (default 4): requires the use of indentation with specified number of spaces.\n\n- `\"tab\"`: requires the use of indentation with tab (`\\t`).\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\n#### space\n\nIf the option is number it means the number of spaces for indentation.\n\n```json\n{\n  \"indent\": [\"error\", 2]\n}\n```\n\nExamples of **incorrect** code for this rule with the `\"2\"` option:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule with the `\"2\"` option:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\n#### tab\n\nIf the option is `\"tab\"` it means using `tab` for indentation.\n\n```json\n{\n  \"indent\": [\"error\", \"tab\"]\n}\n```\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\u003c/body\u003e\n\u003c/html\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003chtml\u003e\n  \u003cbody\u003e\n    \u003c!-- tab --\u003e\n  \u003c/body\u003e\n  \u003c!-- tab --\u003e\n\u003c/html\u003e\n```\n","path":"/docs/rules/indent"}},{"doc":{"title":"no-extra-spacing-attrs","markdown":"\n# no-extra-spacing-attrs\n\nDisallow extra spaces around attributes.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-extra-spacing-attrs\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of extra spaces around attributes.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003c!-- an extra space between attributes --\u003e\n\u003cdiv foo=\"foo\" bar=\"bar\"\u003e\u003c/div\u003e\n\n\u003c!-- an extra space between tag start and attribute --\u003e\n\u003cdiv foo=\"foo\"\u003e\u003c/div\u003e\n\n\u003c!-- an extra space between tag end and attribute --\u003e\n\u003cdiv foo=\"foo\"\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv foo=\"foo\" bar=\"bar\"\u003e\u003c/div\u003e\n\u003cdiv foo=\"foo\"\u003e\u003c/div\u003e\n```\n\n## Options\n\n- `enforceBeforeSelfClose` (default: false): enforce one space before self closing (`/\u003e`)\n\nExamples of **incorrect** code for this rule with the default `{ \"enforceBeforeSelfClose\": true }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg src=\"foo.png\"  /\u003e\n\n\u003cimg src=\"foo.png\"/\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for this rule with the default `{ \"enforceBeforeSelfClose\": true }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cimg src=\"foo.png\" /\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n","path":"/docs/rules/no-extra-spacing-attrs"}},{"doc":{"title":"no-multiple-empty-lines","markdown":"\n# no-multiple-empty-lines\n\nDisallow use of multiple empty lines.\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-multiple-empty-lines\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows the use of empty lines which exceeded the maximum lines allowed.\n\n### Options\n\n- `max` (default 2): enforces a maximum number of consecutive empty lines.\n\nExamples of **incorrect** code for this rule with the default `{ \"max\": 2 }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n\n\n\n\u003cdiv id=\"bar\"\u003e\u003c/div\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n\nExamples of **correct** code for this rule with the default `{ \"max\": 2 }` option:\n\n\u003c!-- prettier-ignore-start --\u003e\n\n```html\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n\n\n\u003cdiv id=\"bar\"\u003e\u003c/div\u003e\n```\n\n\u003c!-- prettier-ignore-end --\u003e\n","path":"/docs/rules/no-multiple-empty-lines"}},{"doc":{"title":"no-trailing-spaces","markdown":"\n# no-trailing-spaces\n\nDisallow trailing white spaces at the end of lines\n\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/no-trailing-spaces\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule disallows trailing white spaces (spaces, tabs) at the end of lines.\n\nExamples of **incorrect** code for this rule:\n\n```html\n\u003cdiv\u003e‚Ä¢‚Ä¢\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule:\n\n```html\n\u003cdiv\u003e\u003c/div\u003e\n```\n","path":"/docs/rules/no-trailing-spaces"}},{"doc":{"title":"quotes","markdown":"\n# quotes\n\n\u003c!-- prettier-ignore-start --\u003e\nEnforce consistent quoting attributes with double(`\"`) or single(`'`)\n## How to use\n\n.eslintrc.js\n\n```js\nmodule.exports = {\n  rules: {\n    \"@html-eslint/quotes\": \"error\",\n  },\n};\n```\n\n## Rule Details\n\nThis rule enforces the consistent use of double(`\"`) or single(`'`) quotes for element attributes.\n\n### Options\n\nThis rule has two options\n\n- `\"double\"` (default): requires the use of double quotes(`\"`).\n- `\"single\"`: requires the use of single quotes(`'`)\n\n#### \"double\"\n\nExamples of **incorrect** code for this rule with the default `\"double\"` option:\n\n\n```html\n\u003cdiv id='foo'\u003e \u003c/div\u003e\n\u003cdiv id=foo\u003e \u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the default `\"double\"` option:\n\n```html\n\u003cdiv id=\"foo\"\u003e \u003c/div\u003e\n\u003cdiv id='containing \"double\" quotes'\u003e\n```\n\n#### \"single\"\n\nExamples of **incorrect** code for this rule with the `\"single\"` option:\n\n```html\n\u003cdiv id=\"foo\"\u003e\u003c/div\u003e\n\u003cdiv id=foo\u003e\u003c/div\u003e\n```\n\nExamples of **correct** code for this rule with the default `\"single\"` option:\n\n```html\n\u003cdiv id='foo'\u003e\u003c/div\u003e\n\u003cdiv id=\"containing 'single' quotes\"\u003e\u003c/div\u003e\n```\n\n## Further reading\n\n- [MDN - Quoting attributes](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/HTML#Quoting_attributes)\n\n\u003c!-- prettier-ignore-end --\u003e\n","path":"/docs/rules/quotes"}}]}]}]},"__N_SSG":true},"page":"/docs/[[...paths]]","query":{"paths":["rules","require-meta-charset"]},"buildId":"9vfwsqTBJOSOD3en28Smg","assetPrefix":"/html-eslint","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>